src/main/java/dev/blaauwendraad/masker/json/KeyContainsMasker.java

```diff
@@ -164,38 +161,103 @@ public byte[] mask(byte[] input) {
             }
             // else, if nothing has matched, continue to the next value
             // it may happen that we don't mask anything if number / boolean masking is disabled or the value is null
-
-            maskingState.backtrackCurrentJsonPath();
         }
 
         ValueMaskingUtil.flushReplacementOperations(maskingState);
 
         return maskingState.getMessage();
     }
 
+    /**
+     * Masks the array element in place.
+     *
+     * @param maskingState the masking state an index prior to the element start index
+     */
+    private void maskArrayElementInPlace(MaskingState maskingState, KeyMaskingConfig keyMaskingConfig) {
+        maskingState.incrementCurrentIndex();
+        skipWhitespaceCharacters(maskingState);
+        boolean maskable = maskValue(maskingState, keyMaskingConfig);
+        if (!maskable) {
+            // non-maskable values
+            skipAllValues(maskingState);
+        }
+    }
+
+    /**
+     * Masks the value in a key/value pair where the current index is on the value start
+     *
+     * @param maskingState     the current masking state in which the value will be masked
+     * @param keyMaskingConfig the masking configuration for the key
+     * @return true if the value is maskable, false otherwise
+     */
+    private boolean maskValue(MaskingState maskingState, KeyMaskingConfig keyMaskingConfig) {
```

Comment:
Clever combination of masking and telling whether it was masked.

Might be good to change the JavaDoc description to include that clearly like: "Checks if the value on the current index in the MaskingState is maskable and masks it if so and return true. Otherwise, doesn't mask and returns false.

Reply:
Thanks! I will fix the javadoc.